python/cugraph/cugraph/dask/cores/core_number.py

```diff
@@ -0,0 +1,179 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dask.distributed import wait, default_client
+from cugraph.dask.common.input_utils import get_distributed_data
+
+import cugraph.dask.comms.comms as Comms
+import dask_cudf
+import cudf
+import warnings
+
+from pylibcugraph.experimental import core_number as \
+    pylibcugraph_core_number
+
+from pylibcugraph import (ResourceHandle,
+                          GraphProperties,
+                          MGGraph
+                          )
+
+
+def call_core_number(sID,
+                     data,
+                     src_col_name,
+                     dst_col_name,
+                     graph_properties,
+                     store_transposed,
+                     num_edges,
+                     do_expensive_check,
+                     degree_type
+                     ):
+
+    handle = Comms.get_handle(sID)
+    h = ResourceHandle(handle.getHandle())
+    srcs = data[0][src_col_name]
+    dsts = data[0][dst_col_name]
+    weights = None
+    if "value" in data[0].columns:
+        weights = data[0]['value']
+
+    mg = MGGraph(h,
+                 graph_properties,
+                 srcs,
+                 dsts,
+                 weights,
+                 store_transposed,
+                 num_edges,
+                 do_expensive_check)
+
+    result = pylibcugraph_core_number(h,
+                                      mg,
+                                      degree_type,
+                                      do_expensive_check)
+
+    return result
+
+
+def convert_to_cudf(cp_arrays):
+    """
+    Creates a cudf DataFrame from cupy arrays from pylibcugraph wrapper
+    """
+    cupy_vertices, cupy_core_number = cp_arrays
+    df = cudf.DataFrame()
+    df["vertex"] = cupy_vertices
+    df["core_number"] = cupy_core_number
+
+    return df
+
+
+def core_number(input_graph,
+                degree_type=None):
+    """
+    Compute the core numbers for the nodes of the graph G. A k-core of a graph
+    is a maximal subgraph that contains nodes of degree k or more.
+    A node has a core number of k if it belongs a k-core but not to k+1-core.
+    This call does not support a graph with self-loops and parallel
+    edges.
+
+    Parameters
+    ----------
+    input_graph : cugraph.graph
+        cuGraph graph descriptor, should contain the connectivity information,
+        (edge weights are not used in this algorithm).
+        The current implementation only supports undirected graphs.
+
+    degree_type: str, Not supported in this release
+        Flag determining whether the core number computation should be based
+        of incoming, outgoing or bidirectional.
+
+        This implementation only supports bidirectional edges.
```

Comment:
```suggestion
    degree_type: str, 
        This option determines if the core number computation should be based
        on input, output, or both directed edges, with valid values being
        "incoming", "outgoing", and "bidirectional" respectively.
        This option is currently ignored in this release, and setting it will
        result in a warning.
```